pkg/generators/forwarding/fluentd/templates.go

```diff
@@ -566,9 +566,20 @@ const pipelineToOutputCopyTemplate = `{{- define "pipelineToOutputCopyTemplate"
     </record>
   </filter>
   {{ end -}}
+  {{ if (eq .Parse "json") -}}
+  <filter **>
+    @type parser
+    key_name message
+    reserve_data yes
+    hash_value_field structured
+    <parse>
+      @type json
+    </parse>
+  <filter>
```

Comment:
@vimalk78 We talked on using `record_modifier` to allow us more flexability

```
 <filter **>
    @type record_modifier
    <record>
      structured      ${require 'json'; JSON.parse(record['message'])}
    </record>
  </filter>
```

Reply:
I used `parse` plugin as suggested by @jcantrill , for two reasons
- It is a generic plugin which can support multiple parsers, and we also want such extensibility in our API
- Originally we either wanted a `message` OR `structured` field after parsing, which needed logic to remove the other. Now requirement is relaxed, so the scope of potential plugins is also increased.

Reply:
@vimalk78 agree ... when we did the inital testing we looked at those two options ... one thing I wonder is performance diffrences ... JSON is known to te expensive to parse ... so if we know that one of the options is much more efficent (again, I don't know) ... maybe it make sense to use the less resource consuming option.

I get the point on future flexability ... agree.

If I remeber correcly there is another small diffrence ... with `parser` we get an empty `structured` field if there is nothing to parse while with `record_modifier` if there is no parsing we have no struct at all ... both are ok, but I think that for customers the second option might be prefered

Reply:
`record_modifier` always adds a `structured` field irrespective of content in `message` field. 

```
structured      ${require 'json'; JSON.parse(record['message'])}
```
The above code will add a field with whatever is the result of `JSON.parse`. It could even be an error message, exception etc, it need not be an empty struct. We will need to handle error conditions more carefully.

I havent checked the performance.`parse` plugin may be slower than `JSON.parse` but has more features.

Reply:
@vimalk78 ok, parser it will be :-)  ... BTW: I also saw another remark from @jcantrill  in the same spirit https://github.com/openshift/cluster-logging-operator/pull/932/files#r587764712